manifests/overlays/dev/kustomization.yaml

```diff
@@ -13,6 +14,9 @@ patches:
       - op: add
         path: /spec/template/spec/containers/0/env/0/value
         value: development
+      - op: add
+        path: /spec/template/spec/serviceAccountName
+        value: service-catalog-k8s-plugin
```

Comment:
Why to use the SA only in dev overlay?

Reply:
My thought process behind not putting this in the prod overlay is that maybe in the future this deployment might be deployed in a cluster other then `smaug`. Thats why I named the SA `service-catalog-k8s-plugin` because it would only be used by the k8s plugin and not as the SA for the pod.

But now when I think about it more, its pretty unlikely that this will be deployed in a different cluster, so I will add this to the base deployment.

Do you also want me to first rename the SA to `service-catalog` or just leave it as is for now and rename it in another PR?

Reply:
Sorry, I don't understand this. Why does it matter which cluster it is deployed to in relation to the Service Account used?

Reply:
What I mean by this is that if we host the service-catalog in a different cluster in the future, and we still want the `k8s plugin` to access the `smaug` cluster it would require a SA and a SA token to use for authentication with the correct privileges (read only access to some resources). This would mean we would need to add the token to the production secret since you can't get the token in other way (as far as I know). This was my initial configuration of the plugin.

In short, it matters in which cluster it is deployed in because if we use `smaug` we don't need the encrypted token in prod overlay, if we don't use `smaug` we need the encrypted token in the prod overlay secret. 

But after your comment and more thinking I realized its probably not worth saving the SA token to the production secret for a small possibility that this will be hosted on a different cluster. And so I also realized its a lot better just to use the mounted token in a pod from `/var/run/secrets/kubernetes.io/serviceaccount/token` if we run the pod with the correct SA (the one with read only access to required resources).

In the first iteration of the plugin configuration I used the mounted token in a pod solution only for the dev cluster, but I think we can use it for booth dev and prod overlays. That is the solution that is implemented right now.

Sorry about my first explanation it was all over the place, hopefully this clears it up more.